LLama.KernelMemory/LLamaSharpTextEmbeddingGenerator.cs

```diff
@@ -112,5 +113,24 @@ public async Task<Embedding> GenerateEmbeddingAsync(string text, CancellationTok
 
         /// <inheritdoc/>
         public int CountTokens(string text) => _embedder.Context.Tokenize(text, special: true).Length;
+
+        /// <summary>
+        /// Get the list of tokens for the input text
+        /// </summary>
+        /// <param name="text">Input string to be tokenized</param>
+        /// <returns>Read-only list of tokens for the input test</returns>
+        /// <remarks>
+        /// It throws if text is null and Includes empty stop token because addBos is left true to be consistent with the CountTokens implementation.</remarks>
+        /// <see cref="CountTokens(string)"/>
+        public IReadOnlyList<string> GetTokens(string text)
+        {
+            var context = _embedder.Context;
+            var embeddings = context.Tokenize(text, special: true);
+            var decoder = new StreamingTokenDecoder(context);
```

Comment:
The purpose of `StreamingTokenDecoder` is to handle the case where one single _character_ is represented by multiple _tokens_ (e.g. some emoji or various non-english characters).

For example let's say the character `A` is represented by the tokens `[1, 2, 3]`. The current code will:
1. Add `1` to the decoder, read an empty string (because `1` on it's own can't be decoded).
2. Add `2` to the decoder, read empty again
3. Add `3` to the decoder, finally read `A`

So the output would be `[ "", "", "A" ]` which obviously is a bit weird!

To be honest I'm not sure what the right fix is here, some tokens simply don't have a string representation - this seems like a fairly badly designed API by the KernelMemory team! We could just stick with the current implementation, that does at least output one string per token, even if some of those strings are empty.

Reply:
I arrived to StreamingTokenDecoder via the _obsolete_ message on the Detokenize method, which was a wrapper for StreamingTokenDecoder anyway. If there's a more appropriate way to decode tokens I'd be glad to switch.

Also, am I correct in referring to the result of the .Tokenize method as embeddings? In retrospect I think _numbers instead of chunks of letters, must be embeddings_ was quite an assumption on my part.